src/mame/drivers/brother_ax145.cpp

```diff
@@ -0,0 +1,352 @@
+// license:BSD-3-Clause
+// copyright-holders:Bartman/Abyss
+
+#include "emu.h"
+#include "emupal.h"
+#include "screen.h"
+#include "speaker.h"
+#include "machine/timer.h"
+#include "cpu/z180/z180.h"
+#include "debug/debugcpu.h"
+#include "sound/beep.h"
+#include "video/hd44780.h"
+
+#pragma region(AX-145)
+
+// 240x18x4 = 960x72
+// -log -debug -window -intscalex 4 -intscaley 4 -resolution 960x72 ax145
+
+//////////////////////////////////////////////////////////////////////////
+// AX-145
+//////////////////////////////////////////////////////////////////////////
+
+/***************************************************************************
+
+Brother AX-145
+198?
+
+Hardware:
+
+Main PCB B482360-1
+========
+
+#7 Hitachi
+HD64180RP6
+6 MHz, 80xx Peripherals
+
+#6 Texas Instruments
+LM393P
+Dual differential comparator, commercial grade
+
+#5 not populated
+µPD23C2001
+
+#2 NEC
+D43256AC-10L
+NEC
+Static CMOS RAM 32,768 x 8-Bit
+
+#3 NEC
+D23C4001EC-172
+UA2849-A
+4MBit Mask ROM for Dictionary
+
+#4 NEC
+D23C1001EC-522
+UA4474-C
+1MBit Mask ROM
+
+#1 Texas Instruments
+SN74HC00N
+Quad 2-Input Positive-NAND Gates
+
+#8 NEC
+D65013-A73
+UA3001-A
+CMOS Gate Array
+
+LCD PCB connected via 12-pin ribbon cable
+========
+Hitachi
+HD44780S-B02
+U17978-A
+Dot Matrix Liquid Crystal Display Controller/Driver
+Custom Font
+(driven in 4-bit mode, R/W connected)
+
+2x Hitachi
+HD66100F
+LCD Driver with 80-Channel Outputs
+
+LCD: 40 characters x 2 lines
+
+// Status:
+// doesn't go further than "SCHREIBWERK ÜBERPRÜFEN" (check printer)
+// needs european font for HD44780
+
+***************************************************************************/
+
+// hw_config:
+// - & 0x1f:
+//    6            => deutsch
+//    12           => francais/nederlands
+//    13           => nederlands
+//    16, 17       => francais/deutsch
+//    8, 9, 14, 21 => espanol
+
+
+class ax145_state : public driver_device
```

Comment:
Please use public/protected/private access levels appropriately, and group data members and member functions within an access level.  The way this is written now is  a mess.

Reply:
better now?